### ISSUE TITLE
Unix sockets

### ISSUE THREAD

**Issue by mmastrac** [wintercg/proposal-sockets-api#16]
Deno currently provides the ability to connect to a unix socket. While there is no standard for unix socket addresses, a few ad-hoc standards do exist for this.

**Comment by dom96** [comment]
Agreed we need to make a decision on this. From what I see in that WhatWG thread it seems there isn't consensus on how to fit into the url standard (though I did mainly skim read it).

**Comment by jasnell** [comment]
I think we can go ahead and allow for flexibility here and just say that in `connect(destination)`, the `destination` MAY be an absolute URL string, and that if it is, it is up to the host implementation to determine if the URL scheme is supported or not.

**Comment by jasnell** [comment]
This would be helpful, btw, if we do end up supporting SNI and ALPN as discussed in other issues. For instance,

`connect('https://example.org')` could automatically set `secureTransport: 'on'`, set the SNI to `'example.org'`, and configure the ALPN for `h1`, etc.
